apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,244 @@
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TempoMonolithicSpec defines the desired state of TempoMonolithic.
+type TempoMonolithicSpec struct {
+	// Storage defines the backend storage configuration
+	//
+	// +kubebuilder:validation:Optional
+	Storage *MonolithicStorageSpec `json:"storage,omitempty"`
+
+	// Ingestion defines the trace ingestion configuration
+	//
+	// +kubebuilder:validation:Optional
+	Ingestion *MonolithicIngestionSpec `json:"ingestion,omitempty"`
+
+	// JaegerUI defines the Jaeger UI configuration
+	//
+	// +kubebuilder:validation:Optional
+	JaegerUI *MonolithicJaegerUISpec `json:"jaegerui,omitempty"`
+
+	// ManagementState defines whether this instance is managed by the operator or self-managed
+	//
+	// +kubebuilder:validation:Optional
+	Management ManagementStateType `json:"management,omitempty"`
+
+	// Observability defines observability configuration for the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Observability *MonolithicObservabilitySpec `json:"observability,omitempty"`
+
+	// ExtraConfig defines any extra (overlay) configuration for components
+	//
+	// +kubebuilder:validation:Optional
+	ExtraConfig *MonolithicExtraConfigSpec `json:"extraConfig,omitempty"`
+}
+
+// MonolithicStorageSpec defines the storage for the Tempo deployment.
+type MonolithicStorageSpec struct {
```

Comment:
One question, Why this have it's own spec and inside only have one structure? Is this for mimic the tempo configuration?

Reply:
Yes, it's to mimic the tempo configuration. I thought maybe tempo has plans to store other things in the future, so I'll keep the same here also.

But I don't have very strong opinions on this, I could remove that extra layer if you like.